crates/configuration/src/configuration.rs

```diff
@@ -75,34 +77,34 @@ pub async fn validate_raw_configuration(
 /// since it consists of a sub-selection of components from the full Configuration, the fields are
 /// borrowed rather than owned.
 #[derive(Debug)]
-pub struct RuntimeConfiguration {
-    pub metadata: metadata::Metadata,
-    pub pool_settings: version1::PoolSettings,
-    pub connection_uri: String,
+pub struct RuntimeConfiguration<'request> {
+    pub metadata: Cow<'request, metadata::Metadata>,
+    pub pool_settings: &'request version2::PoolSettings,
+    pub connection_uri: &'request str,
     pub isolation_level: version2::IsolationLevel,
     pub mutations_version: Option<metadata::mutations::MutationsVersion>,
 }
 
 /// Apply the common interpretations on the Configuration API type into an RuntimeConfiguration.
-pub fn as_runtime_configuration(config: &Configuration) -> RuntimeConfiguration {
+pub fn as_runtime_configuration(config: &Configuration) -> RuntimeConfiguration<'_> {
```

Comment:
I'm not opposed to us using Cow per se, but could we have achieved the same (i.e., avoiding cloning) with ownership:

```
pub fn as_runtime_configuration(config: Configuration) -> RuntimeConfiguration {
```

.. supposing the caller of this function doesn't need to use its `config` value again. I don't remember if it does though.

Reply:
Unfortunately we get a reference, and this is baked into ndc-hub.

Reply:
Oh well, in that case, bring in the COWs